nomad/state/state_store.go

```diff
@@ -1772,105 +1763,6 @@ func (s *StateStore) CSIVolumeDeregister(index uint64, ids []string) error {
 	return nil
 }
 
-// upsertJobCSIPlugins is called on UpsertJob and maintains the csi_plugin index of jobs
-func (s *StateStore) upsertJobCSIPlugins(index uint64, job *structs.Job, txn *memdb.Txn) error {
```

Comment:
The commit messages for this PR don't really make it clear to me why we're pulling this out. Can you expand on that?

Reply:
I added this to the commit message:

> The CSI Plugin API will map plugins to allocations, which allows plugins to be defined by jobs in many configurations. In particular, multiple plugins can be defined in the same job, and multiple jobs can be used to define a single plugin.

> Because we now map the allocation context directly from the node, it's no longer necessary to track the jobs associated with a plugin directly.